File: 0883-projection-area-of-3d-shapes/0883-projection-area-of-3d-shapes.cpp
class Solution {
public:
    int projectionArea(vector<vector<int>>& grid) {
        int ans = 0;
        int maxA, maxB;
        for(int r = 0; r < grid.size(); r++){
            maxA = -1, maxB = -1;
            for(int c = 0; c < grid[r].size(); c++){
                if(grid[r][c] != 0) ans++;
                maxA = max(maxA, grid[r][c]);
                maxB = max(maxB, grid[c][r]);
            }
            ans += (maxA + maxB);
        }
        return ans;
        //  grid의 행과 열의 길이가 같기 때문에 변수로 하나 저장하고 사용했으면 더 빨랐을듯
    }
};

File: 0883-projection-area-of-3d-shapes/0883-projection-area-of-3d-shapes.cs
public class Solution {
    public int ProjectionArea(int[][] grid) {
        int ans = 0;
        int N = grid.Length;
        for(int r = 0; r < N; r++){
            int maxR = -1;
            int maxC = -1;
            for(int c = 0; c < N; c++){
                if(grid[r][c] > 0) ans++;
                maxR = Math.Max(maxR, grid[r][c]);
                maxC = Math.Max(maxC, grid[c][r]);
            }
            ans += (maxR + maxC);
        }
        return ans;
    }
}